static/index.html

```diff
@@ -94,122 +97,168 @@
+                                    aria-hidden="true">&times;</span></button>
+                            <h4>Save as...</h4>
+                        </div>
+                        <div class="modal-body">
+                            <label>Filename: <input class="save-filename" style="width: 30em;" type="text"></label>
+                        </div>
+                        <div class="modal-footer">
+                            <a href="#" data-dismiss="modal" class="btn btn-default">Cancel</a>
+                            <a href="#" class="btn btn-primary save">Save</a>
+                        </div>
                     </div>
-                </form>
-            </div>
-        </div>
-        <div class="row">
-            <div class="col-md-6 editor">
-                <div class="topbar indented">
-                    Code editor
-                    <span class="pull-right input-group input-group-sm">
-                        <button accesskey="a" title="Automatically build when you stop typing"
-                                class="btn btn-default btn-sm autocompile active">
-                            <span class="glyphicon glyphicon-play-circle"></span>
-                        </button>
-                    </span>
                 </div>
-                <textarea>// Type your code here, or load an example.</textarea>
-            </div>
-            <div class="col-md-6 asm">
-                <div class="topbar indented">Assembly output</div>
-                <textarea>Awaiting</textarea>
-            </div>
-        </div>
-        <div class="row result">
-            <div class="topbar">Compiler output &mdash; <span class="compilerVersion"></span></div>
-            <div class="output">
-                <div class="template"></div>
             </div>
         </div>
-    </section>
 
-    <div class="modal fade" id="saveDialog">
-        <div class="modal-dialog">
-            <div class="modal-content">
-                <div class="modal-header">
-                    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span
-                            aria-hidden="true">&times;</span></button>
-                    <h4>Save as...</h4>
+
+        <ul id="draggablePanelList" class="list-unstyled">
+            <li class="panel panel-primary col-md-4" id="slotTemplate">
+            <div class="panel-heading"><span class="title">Slot template (should *not* be visible)</span><span class="closeButton">[X]</span></div>
+            <div class="panel-body">
+                <div class="params">
+                    <form class="well well-sm form-inline">
+                        <table>
+                            <tbody>
+                            <tr>
+                                <td class="dropdown compilers-container">
+                                    <button accesskey="c" type="button" title="Compiler to use (alt-C)" class="btn btn-default dropdown-toggle" data-toggle="dropdown"
```

Comment:
I think we may have to lose the accesskey here and below...it doesn't work with multiple elements sharing the same key (as each slot will define the same access keys). If we want to keep the functionality we'll have to intercept the keys ourselves.